storage/pebble/registers.go

```diff
@@ -22,6 +23,31 @@ type Registers struct {
 
 var _ storage.RegisterIndex = (*Registers)(nil)
 
+func NewRegistersWithPath(dir string) (*Registers, *pebble.DB, error) {
+	db, err := initPebbleDB(dir)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to initialize pebble db: %w", err)
+	}
+	registers, err := NewRegisters(db)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to initialize registers: %w", err)
```

Comment:
what if we return the db here since it was opened successfully.

That gives the caller the chance to do something like
```golang
registers, db, err := pebbleStorage.NewRegistersWithPath(builder.RegistersDBPath)
if err != nil {
	if !errors.Is(err, storage.ErrNotBootstrapped) {
		return nil, err
	}

	// bootstrap

	registers, err = pebbleStorage.NewRegisters(db)
	if err != nil {
		return nil, err
	}
}
```

if not, we need to close it before returning.

Reply:
Good catch